core/src/main/scala/cats/syntax/option.scala

```diff
@@ -394,4 +396,9 @@ object OptionOps {
     def apply[E](ifEmpty: => E)(implicit F: ApplicativeError[F, _ >: E]): F[A] =
       ApplicativeError.liftFromOption(oa, ifEmpty)
   }
+
+  final private[syntax] class ToInvalidAPartiallyApplied[F[_], A](private val oa: Option[A]) extends AnyVal {
```

Comment:
Just passing through, note that we don't get any MiMa support on private classes, even though this exposes public methods.

The fact this even works is probably a manifestation of https://github.com/scala/bug/issues/6794.

Reply:
Yes, I agree – good catch btw. Since `ToInvalidAPartiallyApplied` is supposed to be returned by a public method, then it would be safer to keep it public as well.

Reply:
Oh, good to know! I have just pushed a commit, does it make sense now? Do you know if we should do the same to `LiftToPartiallyApplied` i.e. make it public and add `AnyVal`?

Reply:
I'm afraid that adding `AnyVal` to existing (and thus released) classes can break binary compatibility.

Reply:
Ok, then I will make the `LiftToPartiallyApplied` public as well, this should be compatible.